api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Proxy-WASM ABI.
+ */
+// NOLINT(namespace-envoy)
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+//
+//  ABI calls from the host into the VM.
+//
+//  These will typically be implemented by a language specific SDK which will provide an API on top
+//  of this ABI e.g. the C++ SDK provides a proxy_wasm_api.h implemenation of the API on top of this
+//  ABI.
+//
+
+// Non-stream calls.
+
+/**
+ * Called when the VM starts for each plugin.
+ * @param root_context_id is an identifier for one or more related plugins and their corresponding
+ * proxy_on_configure(), proxy_on_done() and proxy_on_delete() calls. It is also provided during
+ * creation of stream context for those plugins.
+ * @param vm_configuration_size is the size of any configuration available via
+ * proxy_get_configuration during the lifetime of this call.
+ * @return non-zero on success and zero on failure (e.g. bad configuration).
+ */
+extern "C" uint32_t proxy_on_start(uint32_t root_context_id, uint32_t vm_configuration_size);
```

Comment:
Nit: `proxy_on_vm_start` (otherwise it sounds like the proxy is being started..)

Reply:
Done.